code/modules/integrated_electronics/_defines.dm

```diff
@@ -0,0 +1,127 @@
+#define IC_INPUT "input"
+#define IC_OUTPUT "output"
+#define IC_ACTIVATOR "activator"
+
+// Pin functionality.
+#define DATA_CHANNEL "data channel"
+#define PULSE_CHANNEL "pulse channel"
+
+// Methods of obtaining a circuit.
+#define IC_SPAWN_DEFAULT			1 // If the circuit comes in the default circuit box and able to be printed in the IC printer.
+#define IC_SPAWN_RESEARCH 			2 // If the circuit design will be available in the IC printer after upgrading it.
+
+// Displayed along with the pin name to show what type of pin it is.
+#define IC_FORMAT_ANY			"\<ANY\>"
+#define IC_FORMAT_STRING		"\<TEXT\>"
+#define IC_FORMAT_CHAR			"\<CHAR\>"
+#define IC_FORMAT_COLOR			"\<COLOR\>"
+#define IC_FORMAT_NUMBER		"\<NUM\>"
+#define IC_FORMAT_DIR			"\<DIR\>"
+#define IC_FORMAT_BOOLEAN		"\<BOOL\>"
+#define IC_FORMAT_REF			"\<REF\>"
+#define IC_FORMAT_LIST			"\<LIST\>"
+
+#define IC_FORMAT_PULSE			"\<PULSE\>"
+
+// Used inside input/output list to tell the constructor what pin to make.
+#define IC_PINTYPE_ANY				/datum/integrated_io
+#define IC_PINTYPE_STRING			/datum/integrated_io/string
+#define IC_PINTYPE_CHAR				/datum/integrated_io/char
+#define IC_PINTYPE_COLOR			/datum/integrated_io/color
+#define IC_PINTYPE_NUMBER			/datum/integrated_io/number
+#define IC_PINTYPE_DIR				/datum/integrated_io/dir
+#define IC_PINTYPE_BOOLEAN			/datum/integrated_io/boolean
+#define IC_PINTYPE_REF				/datum/integrated_io/ref
+#define IC_PINTYPE_LIST				/datum/integrated_io/list
+
+#define IC_PINTYPE_PULSE_IN			/datum/integrated_io/activate
+#define IC_PINTYPE_PULSE_OUT		/datum/integrated_io/activate/out
+
+// Data limits.
+#define IC_MAX_LIST_LENGTH			200
+//some colors
+#define COLOR_WHITE   			"#FFFFFF"
+#define COLOR_SILVER  			"#C0C0C0"
+#define COLOR_GRAY    			"#808080"
+#define COLOR_BLACK   			"#000000"
+#define COLOR_RED     			"#FF0000"
+#define COLOR_MAROON 			"#800000"
+#define COLOR_YELLOW  			"#FFFF00"
+#define COLOR_OLIVE  			"#808000"
+#define COLOR_LIME   			"#00FF00"
+#define COLOR_GREEN   			"#008000"
+#define COLOR_CYAN    			"#00FFFF"
+#define COLOR_TEAL    			"#008080"
+#define COLOR_BLUE    			"#0000FF"
+#define COLOR_NAVY    			"#000080"
+#define COLOR_PINK    			"#FF00FF"
+#define COLOR_PURPLE  			"#800080"
+#define COLOR_ORANGE  			"#FF9900"
+#define COLOR_LUMINOL 			"#66FFFF"
+#define COLOR_BEIGE 			"#CEB689"
+#define COLOR_BLUE_GRAY 		"#6A97B0"
+#define COLOR_BROWN 			"#B19664"
+#define COLOR_DARK_BROWN 		"#917448"
+#define COLOR_DARK_ORANGE 		"#B95A00"
+#define COLOR_GREEN_GRAY 		"#8DAF6A"
+#define COLOR_RED_GRAY 			"#AA5F61"
+#define COLOR_PALE_BLUE_GRAY	"#8BBBD5"
+#define COLOR_PALE_GREEN_GRAY 	"#AED18B"
+#define COLOR_PALE_RED_GRAY		"#CC9090"
+#define COLOR_PALE_PURPLE_GRAY	"#BDA2BA"
+#define COLOR_PURPLE_GRAY 		"#A2819E"
+#define COLOR_RED_LIGHT         "#FF3333"
+#define COLOR_DEEP_SKY_BLUE     "#00e1ff"
+
+/proc/iscrowbar(O)
+	if(istype(O, /obj/item/crowbar))
+		return 1
+	return 0
+
+var/list/all_integrated_circuits = list()
```

Comment:
`GLOBAL_LIST_EMPTY(all_integrated_circuits)`

You'll have to change all instances of `all_integrated_circuits` to `GLOB.all_integrated_circuits`